### PR TITLE
draft: feat: reuse file from NIC repo

### PR DESCRIPTION
**Pull request by Jcahilltorre**
### Proposed changes
Changes to the hugo config to reuse

https://github.com/nginxinc/kubernetes-ingress/blob/main/deployments/helm-chart/README.md

as part of a hugo content file in
https://github.com/nginxinc/kubernetes-ingress/blob/main/docs/content/installation/installation-with-helm.md

Changes made and why:

go.mod: Added the kubernetes repo as a module to be able to access the files in `deployments/`

config.toml: mounted the `deployments/helm-chart/` path under the `content/` folder (as readfile can only access the files inside the docs/content folder)

sidebar.html: customized the sidebar shortcode to ignore the `/content/remote` files. Even if the files inside are not being processed by hugo for the build, it was still showing on the sidebar due to how this shortcode crawls the content folder

content.htlml: shortcode to use readfile

### CODE REVIEW

**Comment by Jcahilltorre** [comment]
Taking this road causes the TOC not to be generated, as the TOC shortcode is unable to see what other shortcodes inside of the content will generate.

image

**Comment by Jcahilltorre** [comment]
Netlify builds are failing:
`7:15:10 PM: Error: failed to download modules: go command failed: go: gitlab.com/f5/nginx/controller/poc/f5-hugo@v0.24.1: reading gitlab.com/f5/nginx/controller/poc/f5-hugo/controller/poc/f5-hugo/go.mod at revision controller/poc/f5-hugo/v0.24.1: unknown revision controller/poc/f5-hugo/v0.24.1`

Even if I have not made any changes related to the theme module.

https://app.netlify.com/sites/nginx-kubernetes-ingress/deploys/637d1ed90db3dc1b96e4721d#L91

File: docs/go.mod [hunk 1]
```diff
@@ -2,4 +2,7 @@ module github.com/nginxinc/kubernetes-ingress/docs
 
 go 1.15
 
-require gitlab.com/f5/nginx/controller/poc/f5-hugo v0.24.1 // indirect
+require (
+	github.com/nginxinc/kubernetes-ingress v1.12.5 // indirect
```
Comment (lucacome) [inline docs/go.mod:6]: why is it importing version `1.12` of the Ingress Controller? 🤔

**Comment by lucacome** [comment]
> Taking this road causes the TOC not to be generated, as the TOC shortcode is unable to see what other shortcodes inside of the content will generate.


in this thread https://discourse.gohugo.io/t/include-an-md-file/1702/16 they seem to suggest a different approach in the last post, but the are other messages about replacing `markdownify` with `safeHTML`.

File: docs/go.mod (hunk 1 above)
Comment (Jcahilltorre) [reply docs/go.mod:6]: it was imported with `latest` and github provided 1.12. It can be forced to a specific version

**Comment by Jcahilltorre** [comment]
> > Taking this road causes the TOC not to be generated, as the TOC shortcode is unable to see what other shortcodes inside of the content will generate.
> 
> in this thread https://discourse.gohugo.io/t/include-an-md-file/1702/16 they seem to suggest a different approach in the last post, but the are other messages about replacing `markdownify` with `safeHTML`.

removing markdownify causes this:

image

**Pull request closed without merging**